### PR TITLE
github: add issue template

### PR DESCRIPTION
**Pull request by jmxnzo**
Add 3 Github issue templates for:

1.  opening bug reports
2.  feature request 
3.  Q&A redirect

### CODE REVIEW

**Comment by katexochen** [comment]
pr title/commit message should be `github: add issue template`

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: Bug report
+description: File a bug report to help us improve
+labels: ["bug", "needs-triage"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        ####
+        Before starting a bug report, please take a look into the [Troubleshooting Documentation](https://docs.edgeless.systems/contrast/troubleshooting) and revise the existing issue.
```
Comment (katexochen) [inline .github/ISSUE_TEMPLATE/bug_report.yml:9]:
```suggestion
        Before issuing a bug report, please take a look at the [Troubleshooting guide](https://docs.edgeless.systems/contrast/troubleshooting).
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        Further please ensure that the provided CLI logs are configured with the `--log-level` command-line flag set to `debug`.
+
+        Thanks for taking the time to fill out this bug report!
+
+  - type: input
+    id: version
+    attributes:
+      label: Contrast Version
+      description: "In which version do you encounter this bug? (Output of `contrast version`) "
```
Comment (katexochen) [inline .github/ISSUE_TEMPLATE/bug_report.yml:18]: There is no `contrast version`, only `contrast --version`. Also that will print a lot, I don't think we want all of that in the field, right?

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      placeholder: "v0.0.0"
+    validations:
+      required: true
+  - type: input
+    id: platform
+    attributes:
+      label: Deployment Platform
+      description: "On which platform do you encounter this bug? (Value of default_platform value in justfile.env)"
```
Comment (katexochen) [inline .github/ISSUE_TEMPLATE/bug_report.yml:26]: There is no justfile.env for users, that's only a developer thing. ;)

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      placeholder: "AKS-CLH-SNP"
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: Issue Description + Logs
+      description: A concise description of what you're experiencing. Please append the CLI logs(`--log-level debug`)  of the encountered issue right here.
```
Comment (katexochen) [inline .github/ISSUE_TEMPLATE/bug_report.yml:34]:
```suggestion
      description: A concise description of what you're experiencing. Please upload logs of CLI (with `--log-level debug`), Coordinator or other components as file.
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: Steps to Reproduce The Behavior
```
Comment (katexochen) [inline .github/ISSUE_TEMPLATE/bug_report.yml:40]: Please revisit all labels: We follow the [microsoft style guide](https://learn.microsoft.com/en-us/style-guide/welcome/) and only capitalize the first word in a headline.

File: .github/ISSUE_TEMPLATE/config.yml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+blank_issues_enabled: true
+contact_links:
+  - name: Having trouble or questions?
```
Comment (katexochen) [inline .github/ISSUE_TEMPLATE/config.yml:3]:
```suggestion
  - name: Having a questions or an idea to discuss?
```

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+      required: true
+  - type: textarea
+    attributes:
+      label: Describe your solution
+      description: In case you have an idea how to implement this feature.
+  - type: checkboxes
+    attributes:
+      label: Would you be willing to implement this feature?
+      options:
+        - label: Yes, I could contribute this feature.
```
Comment (katexochen) [inline .github/ISSUE_TEMPLATE/feature_request.yml:18]:
```suggestion
```

**Review by katexochen — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 1 above)
Comment (burgerdev) [reply .github/ISSUE_TEMPLATE/bug_report.yml:9]: How about 

> Please take a look at ... before reporting a bug.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 8]
```diff
@@ -0,0 +1,41 @@
+name: Bug report
+description: File a bug report to help us improve
+labels: ["bug", "needs-triage"]
```
Comment (burgerdev) [inline .github/ISSUE_TEMPLATE/bug_report.yml:3]:
```suggestion
labels: ["bug", "needs: triage"]
```

... and the label needs to be created, I guess.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (burgerdev) [reply .github/ISSUE_TEMPLATE/bug_report.yml:18]: Consider adding instructions for reporting unreleased bugs.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 6 above)
Comment (burgerdev) [reply .github/ISSUE_TEMPLATE/config.yml:3]: Suggestion has a typo

File: .github/ISSUE_TEMPLATE/config.yml
Comment (burgerdev) [inline .github/ISSUE_TEMPLATE/config.yml:1]: Why is this in `config.yml` and not in, say, `q-and-a.yml`?

Comment (jmxnzo) [reply .github/ISSUE_TEMPLATE/config.yml:1]: I reused the templates of Constellation, thus the file names remained the same. I changed it to q-and-a.yml for now!

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 8 above)
Comment (katexochen) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Label still wasn't created. Maybe we can reuse `needs: assignee`?

File: .github/ISSUE_TEMPLATE/config.yml
Comment (katexochen) [reply .github/ISSUE_TEMPLATE/config.yml:1]: Actually, is has to be named config.yml, see https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser

This isn't another template but the config file of the template chooser.

**Review by katexochen — CHANGES_REQUESTED** [review]
see unresolved conversations

File: .github/ISSUE_TEMPLATE/config.yml
Comment (burgerdev) [reply .github/ISSUE_TEMPLATE/config.yml:1]: Sorry, my bad, did not read carefully enough.

Comment (jmxnzo) [reply .github/ISSUE_TEMPLATE/config.yml:1]: No worries me too, i changed it back again!

**Review by katexochen — APPROVED** [review]

**Pull request merged**